File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Progression.java
package com.tdevilleduc.urthehero.back.model;

import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotNull;

@Entity
@RequiredArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class Progression {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    private Integer id;
    @NotNull
    private Integer storyId;
    @NotNull
    private Integer personId;
    @NotNull
    private Integer actualPageId;

}

File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Progression.kt
package com.tdevilleduc.urthehero.back.model

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.validation.constraints.NotNull

@Entity
data class Progression(var storyId: Int,
                       var userId: Int,
                       var actualPageId: Int) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: @NotNull Int? = null
}